compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -99,6 +99,22 @@ impl Attribute {
         }
     }
 
+    pub fn is_path(&self, name: &[Symbol]) -> bool {
+        match &self.kind {
+            AttrKind::Normal(normal) => {
+                normal.item.path.segments.len() == name.len()
+                    && normal
+                        .item
+                        .path
+                        .segments
+                        .iter()
+                        .zip(name)
+                        .all(|(s, n)| s.ident == Ident::with_dummy_span(*n))
```

Comment:
This equality is weird, and relies on the fact that idents are equal only if their symbols are equal and their span contexts are identical:
https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_span/symbol.rs.html#1795-1800

If you're just trying to compare the symbols against `name`, then try
```suggestion
                        .all(|(s, n)| s.ident.name == *n)
```